Parser/parsetok.c

```diff
@@ -38,10 +38,11 @@ static int
 growable_comment_array_add(growable_comment_array *arr, int lineno, char *comment) {
     if (arr->num_items >= arr->size) {
         arr->size *= 2;
-        arr->items = realloc(arr->items, arr->size * sizeof(*arr->items));
-        if (!arr->items) {
+        void *const new_items_array = realloc(arr->items, arr->size * sizeof(*arr->items));
```

Comment:
I don't see the purpose of "const" here. I suggest to remove it. realloc() return type is `void *`.

Reply:
I mean, hey, if you guys don't like const correctness, I can do that. Thankfully, I have enough control over my OCD that it works for me :)

Reply:
Actually, while I'm here, should I also *not* double the array size on the failure path?

And should I squash the multiple commits?

Reply:
"hey, if you guys": please use a more inclusive expressions, see https://heyguys.cc/ ;-)

"don't like const correctness": it's assigned to arr->items 3 lines below but growable_comment_array.items isn't constant. I don't see the value of using const here.

You should read https://theartofmachinery.com/2019/08/12/c_const_isnt_for_performance.html "Why const Doesn't Make C Code Faster" :-)

Reply:
> should I also not double the array size on the failure path?

Right, arr->size should only be modified on success. Replace maybe `realloc(arr->items, arr->size * sizeof(*arr->items))` with `realloc(arr->items, (arr->size * 2) * sizeof(*arr->items))`.

> And should I squash the multiple commits?

That's harmful for reviewers. I prefer to see new commits.

We always squash all commits of a PR when it's merged.

Reply:
Let me explain differently: your usage of const is correct. But IMHO it's not worth it, and I see it more as an annoyance for readers.